manifests/init.pp

```diff
@@ -929,11 +952,9 @@
         mode      => '0640',
         content   => template('datadog_agent/datadog.yaml.erb'),
         show_diff => false,
+        notify    => Service[$datadog_agent::params::service_name],
         require   => File['/etc/datadog-agent'],
       }
-      if ! $_agent_managed_by_installer {
-        File['/etc/datadog-agent/datadog.yaml']  ~> Service[$datadog_agent::params::service_name]
-      }
```

Comment:
Question ❓ :
Isn't it important to keep these lines so that the service is refreshed whenever `datadog.yaml` is updated ?

Reply:
Answered on slack -- 
```puppet
    class { 'datadog_agent::service' :
      # Declare service for agent managed by installer with installer flavor
      agent_flavor     => 'datadog-installer',
```
solves it